api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImpl.java

```diff
@@ -42,43 +55,97 @@ public Location getLocationByUuid(String uuid) {
 	}
 	
 	@Override
-	public Collection<Location> findLocationByName(String name) {
-		return sessionFactory.getCurrentSession().createCriteria(Location.class).add(eq("name", name)).list();
+	public Collection<Location> searchForLocations(StringOrListParam name, StringOrListParam city, StringOrListParam country,
+	        StringOrListParam postalCode, StringOrListParam state, TokenOrListParam tag, ReferenceOrListParam parent,
+	        SortSpec sort) {
+		
+		Criteria criteria = this.sessionFactory.getCurrentSession().createCriteria(Location.class);
+		handleBooleanProperty("retired", false).ifPresent(criteria::add);
+		
+		handleName(criteria, name);
+		handleCity(criteria, city);
+		handleCountry(criteria, country);
+		handlePostalCode(criteria, postalCode);
+		handleState(criteria, state);
+		handleTag(criteria, tag);
+		handleParentLocation(criteria, parent);
+		handleSort(criteria, sort);
+		
+		return criteria.list();
 	}
 	
 	@Override
-	public Collection<Location> findLocationsByCity(String city) {
-		return sessionFactory.getCurrentSession().createCriteria(Location.class).add(eq("cityVillage", city)).list();
+	public List<LocationAttribute> getActiveAttributesByLocationAndAttributeTypeUuid(Location location,
+	        String locationAttributeTypeUuid) {
+		return (List<LocationAttribute>) sessionFactory.getCurrentSession().createCriteria(LocationAttribute.class)
+		        .createAlias("location", "l", JoinType.INNER_JOIN, eq("l.id", location.getId()))
+		        .createAlias("attributeType", "lat").add(eq("lat.uuid", locationAttributeTypeUuid)).add(eq("voided", false))
+		        .list();
 	}
 	
-	@Override
-	public Collection<Location> findLocationsByCountry(String country) {
-		return sessionFactory.getCurrentSession().createCriteria(Location.class).add(eq("country", country)).list();
+	private void handleName(Criteria criteria, StringOrListParam namePattern) {
+		if (namePattern != null) {
+			handleOrListParam(namePattern, (givenName) -> propertyLike("name", givenName)).ifPresent(criteria::add);
```

Comment:
`givenName` really only applies to things with a `PersonName` (OpenMRS) or `HumanName` (FHIR). This should just be called `name`, i.e., it gets called "givenName" to distinguish it from "familyName" (it makes a bit more sense in French where they're, respectively, "prénom" and "nom"). Very European-parochial.